src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -285,5 +378,31 @@ public override unsafe bool TryEncodeUnicodeScalar(int unicodeScalar, char* buff
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool NeedsEscaping(char value) => value > LastAsciiCharacter || AllowList[value] == 0;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int NeedsEscaping(Vector128<sbyte> sourceValue)
```

Comment:
> @ahsonkhan please help me figure out what causes the [build-failure]

You are seeing the failure particularly on the netfx leg. Vector128 is not available there (alone with the Sse2/Ssse3/other hardware intrinsics. Those are only available on netcoreapp.

You need to put this code under the `#if NETCOREAPP` block (similar to the other code that references those types). Or move it to a file that is conditionally included like Sse2Helper.cs:

https://github.com/dotnet/corefx/blob/bd18c12df9e5dd365a8b708cc2e7e0488b8177c1/src/System.Text.Encodings.Web/src/System.Text.Encodings.Web.csproj#L25-L27

Reply:
Ahh, thanks! So I know to check this with `./build.sh -f netfx`.
Will fix that.

Reply:
For the failed tests: when `utf8Text` is empty, we get a nullptr from `fixed` (so far nothing new to me), but then at subtracting from this nullptr, it will overflow, so the vectorized path is taken and loading the vector it crashed. Good that there are tests -- and I've learned that nullptr need more care than I thought.

Reply:
Yep, the current implementation in master is safe against the null ptr access because all such operations are within length checks. Hence the tests :)